src/dwn/models/record.js

```diff
@@ -0,0 +1,244 @@
+import { DataStream, DwnConstant, Encoder } from '@tbd54566975/dwn-sdk-js';
+import { ReadableWebToNodeStream } from 'readable-web-to-node-stream';
+
+import { isReadableWebStream } from '../dwn-utils.js';
+
+export class Record {
+  #dwn;
+
+  #author;
+  #contextId;
+  #descriptor;
+  #recordId;
+  #target;
+
+  #encodedData = null;
+  #isFrozen = false;
+  #readableStream = null;
+
+  constructor(dwn, options = { }) {
+    this.#dwn = dwn;
+
+    // RecordsWriteMessage properties.
+    const { author, contextId = undefined, descriptor, recordId = null, target } = options;
+    this.#contextId = contextId;
+    if (descriptor?.data) delete descriptor.data;
```

Comment:
can we do `delete this.#descriptor.data;` below instead?  then we dont have to check for anything

Reply:
Incredibly minor nit but sure...
Addressed in 9786d34e0dedaeb68c519b78844cd613042f819b